backend/cfg/merge_straightline_blocks.ml

```diff
@@ -86,13 +86,15 @@ let rec merge_blocks (modified : bool) (cfg_with_layout : Cfg_with_layout.t) :
             b2_block.terminator
               <- { b2_block.terminator with desc = Cfg_intf.S.Never };
             b2_block.exn <- None;
-            true)
-          else merged
-        | _ -> merged)
-      cfg.blocks false
+            Label.Set.add b2_label acc)
+          else acc
+        | _ -> acc)
+      cfg.blocks Label.Set.empty
   in
-  if merged then merge_blocks true cfg_with_layout else modified
+  if not (Label.Set.is_empty new_removed)
+  then merge_blocks (Label.Set.union new_removed removed) cfg_with_layout
+  else removed
 
 let run (cfg_with_layout : Cfg_with_layout.t) : unit =
-  let modified = merge_blocks false cfg_with_layout in
-  if modified then Eliminate_dead_code.run_dead_block cfg_with_layout
+  merge_blocks Label.Set.empty cfg_with_layout
+  |> Cfg_with_layout.remove_blocks cfg_with_layout
```

Comment:
I don't think it has the same effect as before;
wouldn't `Eliminate_dead_code.run_dead_block`
remove more blocks?

Reply:
That first call you mention was what I was afraid of,
but I didn't examine the call sites; the added comment
is indeed useful.